### PR TITLE
Perf 3545 Migrate Genny Workloads doing Fsync via the RunCommandActor to the QuiesceActor

### PR DESCRIPTION
**Pull request by SimonEismann**
This pull requests migrates all existing Genny workload that use the RunCommand actor to perform an Fsync to using the newer QuiesceActor. This affects the following workloads:

* TimeSeriesLastpoint
* CreateBigIndex
* ExternalSort
* GraphLookup
* GraphLookupWithOnlyUnshardedColls
* Lookup
* LookupWithOnlyUnshardedColls
* SlidingWindow
* UnionWith
* MixedWorkloadsGenny
* MixedWorkloadsGennyRateLimited
* MixedWorkloadsGennyStress

To verify that these changes do not impact performance, I ran them in this [patch](https://spruce.mongodb.com/version/6389ee7aa4cf477e25f9f934). This showed some ~20% deterioration for some parts of the MixedWorkloadsGenny workload. Upon further inspection, there were several similar outliers in the past so I reran this workload. 

The second execution now shows no significant deterioration in performance: [PerfAnalyzer](https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?evergreen_version=6389ee7aa4cf477e25f9f934&evergreen_base_version=sys_perf_552afb106b2cdc485db67087ce24e7d33e8cbabb&metrics_selection=Latency99thPercentile||ops_per_sec||Latency50thPercentile||OperationThroughput||tpmC||NewOrders||Duration||average_read_latency_us||95th_read_latency_us||99th_read_latency_us&percent_filter=0||100&z_filter=0||10)

### CODE REVIEW

File: src/workloads/execution/CreateBigIndex.yml [hunk 1]
```diff
@@ -23,21 +23,30 @@ Actors:
...
+  - *Nop
 
-# Phase 2: Lower the memory limit for spilling to disk so that it occurs more often.
+# Phase 3: Lower the memory limit for spilling to disk so that it occurs more often.
 - Name: Setup
   Type: RunCommand
   Threads: 1
   Phases:
   - *Nop
+  - *Nop
```
Comment (jimoleary) [inline src/workloads/execution/CreateBigIndex.yml:46]: In order to not change the results, I think you should probably keep the ```LimitIndexBuildMemoryUsageCommand``` in the second phase. 

Or at least run a patch to ensure the results are the same.

File: src/workloads/query/Lookup.yml [hunk 2]
```diff
@@ -77,17 +77,13 @@ Actors:
   - *Nop
 
 - Name: Quiesce
-  Type: RunCommand
+  Type: QuiesceActor
   Threads: 1
+  Database: *Database
```
Comment (jimoleary) [inline src/workloads/query/Lookup.yml:82]: different database

File: src/workloads/query/LookupWithOnlyUnshardedColls.yml [hunk 3]
```diff
@@ -34,16 +34,12 @@ Actors:
   - *Nop
 
 - Name: Quiesce
-  Type: RunCommand
+  Type: QuiesceActor
   Threads: 1
+  Database: *Database
```
Comment (jimoleary) [inline src/workloads/query/LookupWithOnlyUnshardedColls.yml:39]: database

File: src/workloads/query/SlidingWindows.yml [hunk 4]
```diff
@@ -26,16 +26,12 @@ Actors:
   - *Nop
 
 - Name: Quiesce
-  Type: RunCommand
+  Type: QuiesceActor
   Threads: 1
+  Database: *db
```
Comment (jimoleary) [inline src/workloads/query/SlidingWindows.yml:31]: different database

File: src/workloads/query/TimeSeriesLastpoint.yml [hunk 5]
```diff
@@ -16,10 +16,6 @@ phasePath: &phasePath ../../phases/query/TimeSeriesLastpoint.yml
 
 # Operations reused in multiple Phases.
 Nop: &Nop {Nop: true}
-Quiesce: &Quiesce
```
Comment (jimoleary) [inline src/workloads/query/TimeSeriesLastpoint.yml:19]: Shouldn't we also change the Quiesce key and use it (or remove it from the source location).

File: src/workloads/query/TimeSeriesLastpoint.yml [hunk 6]
```diff
@@ -422,6 +418,36 @@ Actors:
         Collection: *coll
         IndexName: *metaAscTimeAsc
 
+- Name: Quiesce
```
Comment (jimoleary) [inline src/workloads/query/TimeSeriesLastpoint.yml:421]: I think this would be better as the following the phases:

```
# Parameterized Quiesce Actor
QuiesceActor:
  Name: {^Parameter: {Name: "Name", Default: "FindOne"}}
  Type: QuiesceActor
  Threads: 1
  Database: {^Parameter: {Name: "Database", Default: *db_name}}
  Phases:
    OnlyActiveInPhases:
      Active: {^Parameter: {Name: "Active", Default: [0, 2, 4, 6, 8, 10]}}
      NopInPhasesUpTo: {^Parameter: {Name: "MaxPhases", Default: 20}}
      PhaseConfig:
        SleepBefore: 1 seconds
        SleepAfter: 1 seconds
        Repeat: 1
```

And then here would look like:

```
- LoadConfig:
    Path: *path
    Key: QuiesceActor
    Parameters:
      Active: [2, 5, 8, 11, 14, 17, 18, 21]
      MaxPhases: 21 # or whatever this is.
```


**Also why are there 2 consecutive quiesces?**

File: src/workloads/execution/CreateBigIndex.yml (hunk 1 above)
Comment (SimonEismann) [reply src/workloads/execution/CreateBigIndex.yml:46]: As per our discussion, I've updated this so both`LimitIndexBuildMemoryUsageCommand ` and the QuiesceActor run in the same phase. 

I've also tested that none of the changes seem to impact performance: [PerfAnalyzer](https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?evergreen_version=6389ee7aa4cf477e25f9f934&evergreen_base_version=sys_perf_552afb106b2cdc485db67087ce24e7d33e8cbabb&metrics_selection=Latency99thPercentile%7C%7Cops_per_sec%7C%7CLatency50thPercentile%7C%7COperationThroughput%7C%7CtpmC%7C%7CNewOrders%7C%7CDuration%7C%7Caverage_read_latency_us%7C%7C95th_read_latency_us%7C%7C99th_read_latency_us&percent_filter=0%7C%7C100&z_filter=0%7C%7C10)

File: src/workloads/query/GraphLookupWithOnlyUnshardedColls.yml [hunk 7]
```diff
@@ -41,16 +41,12 @@ Actors:
   - *Nop
 
 - Name: Quiesce
-  Type: RunCommand
+  Type: QuiesceActor
   Threads: 1
+  Database: *Database
```
[the remark this answers is not in the document]
Comment (SimonEismann) [reply src/workloads/query/GraphLookupWithOnlyUnshardedColls.yml:46]: updated

File: src/workloads/query/Lookup.yml (hunk 2 above)
Comment (SimonEismann) [reply src/workloads/query/Lookup.yml:82]: updated

File: src/workloads/query/LookupWithOnlyUnshardedColls.yml (hunk 3 above)
Comment (SimonEismann) [reply src/workloads/query/LookupWithOnlyUnshardedColls.yml:39]: updated

File: src/workloads/query/SlidingWindows.yml (hunk 4 above)
Comment (SimonEismann) [reply src/workloads/query/SlidingWindows.yml:31]: updated

File: src/workloads/query/TimeSeriesLastpoint.yml (hunk 5 above)
Comment (SimonEismann) [reply src/workloads/query/TimeSeriesLastpoint.yml:19]: As far as I understand, I've removed the original quiesce definition here:
src/phases/query/TimeSeriesLastpoint.yml

File: src/workloads/query/TimeSeriesLastpoint.yml (hunk 6 above)
Comment (SimonEismann) [reply src/workloads/query/TimeSeriesLastpoint.yml:421]: Implemented the proposed changes, this significantly improves readability. 

Regarding the two consecutive quiesces, this does seem weird but is also the case in the original workload:

https://github.com/mongodb/genny/blob/4c87a4f5f743fd758b0786bf0e7d0484c2853063/src/workloads/query/TimeSeriesLastpoint.yml#L377
https://github.com/mongodb/genny/blob/4c87a4f5f743fd758b0786bf0e7d0484c2853063/src/workloads/query/TimeSeriesLastpoint.yml#L341

Should we reach out to the test authors about this or just keep as is?

**Comment by SimonEismann** [comment]
I've pushed a commit to incorporate the feedback so far, will run a patch to ensure I didn't break anything while incorporating the feedback.

File: src/workloads/query/TimeSeriesLastpoint.yml (hunk 5 above)
Comment (jimoleary) [reply src/workloads/query/TimeSeriesLastpoint.yml:19]: Thanks.

File: src/phases/query/TimeSeriesLastpoint.yml [hunk 8]
```diff
@@ -92,3 +83,17 @@ DropIndex:
     OperationCommand:
       dropIndexes: {^Parameter: {Name: "Collection", Default: "Collection0"}}
       index: {^Parameter: {Name: "IndexName", Default: ""}}
+
+QuiesceActor:
+  Name: {^Parameter: {Name: "Name", Default: "FindOne"}}
```
Comment (jimoleary) [inline src/phases/query/TimeSeriesLastpoint.yml:88]: [nit] Default name should probably be QuiesceActor

File: src/phases/query/TimeSeriesLastpoint.yml [hunk 9]
```diff
@@ -92,3 +83,17 @@ DropIndex:
...
+  Type: QuiesceActor
+  Threads: 1
+  Database: {^Parameter: {Name: "Database", Default: admin}}
+  Phases:
+    OnlyActiveInPhases:
+      Active: {^Parameter: {Name: "Active", Default: [0, 2, 4, 6, 8, 10]}}
```
Comment (jimoleary) [inline src/phases/query/TimeSeriesLastpoint.yml:94]: Can we make the default ```[2, 5, 8, 11, 14, 17, 18, 21]``` and then remove Active from the call site?

File: src/workloads/query/TimeSeriesLastpoint.yml [hunk 10]
```diff
@@ -422,6 +418,13 @@ Actors:
         Collection: *coll
         IndexName: *metaAscTimeAsc
 
+- LoadConfig:
+    Path: *phasePath
+    Key: QuiesceActor
+    Parameters:
+      Active: [2, 5, 8, 11, 14, 17, 18, 21]
```
Comment (jimoleary) [inline src/workloads/query/TimeSeriesLastpoint.yml:425]: maybe make these the defaults?

**Review by jimoleary — COMMENTED** [review]
I have a couple of small comments. But it is broadly lgtm.

Are there any performance changes ?

File: src/phases/query/TimeSeriesLastpoint.yml (hunk 9 above)
Comment (SimonEismann) [reply src/phases/query/TimeSeriesLastpoint.yml]: Definitely better than having random defaults, fixed

**Comment by SimonEismann** [comment]
I've added the changes, for the initial version there were no detected performance changes
image

Patch for the version that includes the feedback is still running and I'll update with the results here :)

**Review by jimoleary — APPROVED** [review]
